packages/babel-parser/src/parser/statement.ts

```diff
@@ -1139,13 +1223,11 @@ export default abstract class StatementParser extends ExpressionParser {
       kind: kind,
       statementStart: this.state.start,
     });
-    node.body = this.parseStatement(
-      context
-        ? context.indexOf("label") === -1
-          ? context + "label"
-          : context
-        : "label",
```

Comment:
The `context` variable is now replaced by an option bag `ParseStatementFlag`. Personally I was constantly confused by the dynamic `context` string, which seems to be an overkill as we are not using that string for error reports.